### ISSUE TITLE
[Optimization] Calculate the projection matrix only when FOV or window size changed.

### ISSUE THREAD

**Issue by malortie** [Solokiller/HL_Tools#132]
The projection matrix appears to be calculated each time the model is drawn.

https://github.com/Solokiller/HL_Tools/blob/597a3a3bdd23daa5eaf2cf32d5c86a5a16cbf16a/src/tools/hlam/graphics/Scene.cpp#L351

https://github.com/Solokiller/HL_Tools/blob/597a3a3bdd23daa5eaf2cf32d5c86a5a16cbf16a/src/tools/hlam/graphics/GraphicsUtils.cpp#L109-L115

In the code, the near and far plane distance values are constants, but the FOV and window size can vary based on user input.

The recommendation would be to store the projection matrix as a variable that would be updated only when the following events occur:

- User changed the field of view
- User resized the window

The projection matrix variable could then be used as follows:

```cpp
glLoadMatrixf( glm::value_ptr( camera->GetProjectionMatrix() ) );
```

**Comment by Solokiller** [comment]
I've provided an alpha build that fixes this issue. Can you test it and see if it's fixed?

https://github.com/Solokiller/HL_Tools/releases/tag/HLAM-V1.1.0-alpha001

**Comment by malortie** [comment]
I tested with the 1.1.0 alpha 002. It should also work in the alpha 001, as the only difference between alpha 002 and 001 is the [fix for the flat shade](https://github.com/Solokiller/HL_Tools/compare/HLAM-V1.1.0-alpha001...HLAM-V1.1.0-alpha002).

For each camera type, I tested the following:

- Resizing the window (width and height).
- Changing the FOV.
- Press 'Default' FOV button to ensure the FOV is correctly reset.

The changes seem to work correctly. I also looked at the new code and it seems to be ok, so I'll close this issue.
